opencensus/common/monitored_resource_util/gcp_metadata_config.py

```diff
@@ -95,6 +95,9 @@ def _initialize_metadata_service(cls):
         if instance_id is not None:
             cls.is_running = True
 
+            if isinstance(instance_id, bytes):
```

Comment:
What about moving the type conversion logic to the function `_get_attribute` below?